### PR TITLE
chore: generate more types

### PR DESCRIPTION
**Pull request by dgozman**
As a result, fix documentation issues:
- Removed `TestStep.data` that is not a thing.
- Added `TestConfig.name` documentation.
- Refined a lot of optional types in the documentation.
- Aligned `test.fail()` and `test.slow()` docs with `test.skip()`.
- Strict set of docs vs types inconsistensies in the generator

### CODE REVIEW

File: docs/src/test-api/class-test.md [hunk 1]
```diff
@@ -1136,17 +1147,15 @@ test('skip in WebKit 2', async ({ page }) => {
...
+Optional description that will be reflected in a test report.
 
 
-## method: Test.slow
 
-Marks a test or a group of tests as "slow". Slow tests will be given triple the default timeout.
+## method: Test.slow#1
 
-Unconditional slow:
+Unconditionally marks a test as "slow". Slow test will be given triple the default timeout.
```
[the remark this answers is not in the document]
Comment (rwoll) [reply docs/src/test-api/class-test.md:1158]: Applies to the other occurrences of this phrasing below.

File: docs/src/test-api/class-testinfo.md [hunk 2]
```diff
@@ -166,41 +166,52 @@ test.afterEach(async ({}, testInfo) => {
 });
 ```
 
-## method: TestInfo.fail
+## method: TestInfo.fail#1
 
-Marks the currently running test as "should fail". Playwright Test ensures that this test is actually failing. This is similar to [`method: Test.fail`].
+Marks the currently running test as "should fail". Playwright Test runs theis tests and ensures that it is actually failing. This is useful for documentation purposes to acknowledge that some functionality is broken until it is fixed. This is similar to [`method: Test.fail#1`].
```
Comment (rwoll) [inline docs/src/test-api/class-testinfo.md:171]:
```suggestion
Marks the currently running test as "should fail". Playwright Test runs this tests and ensures that it is actually failing. This is useful for documentation purposes to acknowledge that some functionality is broken until it is fixed. This is similar to [`method: Test.fail#1`].
```

File: utils/generate_types/overrides-testReporter.d.ts [hunk 3]
```diff
@@ -17,63 +17,16 @@
 import type { FullConfig, FullProject, TestStatus, TestError } from './test';
 export type { FullConfig, TestStatus, TestError } from './test';
```
Comment (rwoll) [inline utils/generate_types/overrides-testReporter.d.ts:19]: To confirm, all these are now generated from docs directly?

**Review by rwoll — APPROVED** [review]
Overall looks good! Left a few suggestions (and some typo fixes).

Cheers! 🎉

File: docs/src/test-api/class-test.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgozman) [reply docs/src/test-api/class-test.md:1158]: `slow` is actually about the timeout, not about being reported as slow.

[the remark this answers is not in the document]
Comment (rwoll) [reply docs/src/test-api/class-test.md:1158]: Ah—I thought it was both! Let's clarify in the docs

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**